xformers/components/attention/csrc/cuda/mem_eff_attention/attention_backward_generic.cu

```diff
@@ -139,13 +139,23 @@ mem_efficient_attention_backward_cutlass(
           kernel_fn, cudaFuncAttributeMaxDynamicSharedMemorySize, smem_bytes);
     }
 
-    auto checkBinaryArchMatches = [&]() {
-      cudaFuncAttributes attr;
-      AT_CUDA_CHECK(cudaFuncGetAttributes(&attr, kernel_fn));
-      return attr.binaryVersion >= Kernel::ArchTag::kMinComputeCapability;
-    };
+    // previous syntax, in comment below, resulted in the error below on windows
+    // error C3495: 'kernel_fn': a simple capture must be a variable
+    // with automatic storage duration declared
+    // in the reaching scope of the lambda
+    // auto checkBinaryArchMatches = [&]() {
+    //   cudaFuncAttributes attr;
+    //   AT_CUDA_CHECK(cudaFuncGetAttributes(&attr, kernel_fn));
+    //   return attr.binaryVersion >= Kernel::ArchTag::kMinComputeCapability;
+    // };
+    // TORCH_INTERNAL_ASSERT(
+    //     checkBinaryArchMatches(),
+    // "Something went wrong in the build process");
+    cudaFuncAttributes attr;
+    AT_CUDA_CHECK(cudaFuncGetAttributes(&attr, kernel_fn));
     TORCH_INTERNAL_ASSERT(
-        checkBinaryArchMatches(), "Something went wrong in the build process");
+        attr.binaryVersion >= Kernel::ArchTag::kMinComputeCapability,
+        "Something went wrong in the build process");
```

Comment:
as this is just a sanity check - can we keep it as it was but under an `#ifdef` guard for windows?

Reply:
https://github.com/facebookresearch/xformers/pull/460/commits/8509baaef125121c4cf7f5476eb4c26e1d0769c6 should do it